server/routes/serviceProviderReferrals/showReferralPresenter.ts

```diff
@@ -39,24 +41,41 @@ export default class ShowReferralPresenter {
     { key: 'Email address', lines: [this.sentBy.email ?? ''] },
   ]
 
-  get interventionDetails(): SummaryListItem[] {
-    // const selectedDesiredOutcomes = this.serviceCategory.desiredOutcomes
-    //   .filter(desiredOutcome => this.sentReferral.referral.desiredOutcomesIds.includes(desiredOutcome.id))
-    //   .map(desiredOutcome => desiredOutcome.description)
-    //
-    // const selectedComplexityLevel = this.serviceCategory.complexityLevels.find(
-    //   complexityLevel => complexityLevel.id === this.sentReferral.referral.complexityLevelId
-    // )
-    //
-    // const complexityLevelText = {
-    //   level: selectedComplexityLevel?.title || 'Level not found',
-    //   text: selectedComplexityLevel?.description || 'Description not found',
-    // }
+  get referralServiceCategories(): ServiceCategory[] {
+    const { serviceCategoryIds } = this.sentReferral.referral
+    return this.intervention.serviceCategories.filter(it => serviceCategoryIds.includes(it.id))
+  }
+
+  serviceCategorySection(serviceCategory: ServiceCategory, tagMacro: (args: TagArgs) => string): SummaryListItem[] {
+    const items: SummaryListItem[] = []
+
+    const complexityLevel = this.getReferralComplexityLevelForServiceCategory(serviceCategory)
+    if (complexityLevel === null) {
+      items.push({ key: 'Complexity level', lines: ['Level not found'] })
+    } else {
+      items.push({
+        key: 'Complexity level',
+        lines: [tagMacro(PresenterUtils.complexityLevelTagArgs(complexityLevel)), complexityLevel.description],
```

Comment:
Does`lines: [complexityLevel === null ? 'Level not found' : tagMacro(PresenterUtils.complexityLevelTagArgs(complexityLevel)), complexityLevel.description]` work here? Would mean we can remove the conditionals and the pushing into an array here, right?

Reply:
i'm gonna leave it for now and when i do the desiredoutcomes i'll look at refactoring